### ISSUE TITLE
invalid zip file generated when compressing empty folder

### ISSUE THREAD

**Issue by dracodoc** [r-lib/zip#2]
@gaborcsardi 
The `files` parameter in `zip` can be a folder according to examples, even the document didn't state so explicitly. 

When this folder is an empty folder, the resulting zip is of size 22 bytes and not a valid zip file.

```r
dir.create(tmp <- tempfile())
zipfile <- tempfile(fileext = ".zip")
# utils::zip(zipfile, tmp)
zip::zip(zipfile, tmp)
zip_list(zipfile)
# [1] filename          compressed_size   uncompressed_size
# <0 rows> (or 0-length row.names)
unzip(zipfile)
# Warning message:
#   In unzip(zipfile) : error 1 in extracting from zip file
```

Compare to the `utils::zip` behavior, which should be desired

```r
dir.create(tmp <- tempfile())
zipfile <- tempfile(fileext = ".zip")
utils::zip(zipfile, tmp)
# adding: var/folders/f3/6cdx4zf94qz95r6d89s_4xcsrcsjqm/T//Rtmp1ZMU5C/file8a76deabf6b/ (stored 0%)
# zip::zip(zipfile, tmp)
zip_list(zipfile)
# filename compressed_size
# 1 var/folders/f3/6cdx4zf94qz95r6d89s_4xcsrcsjqm/T//Rtmp1ZMU5C/file8a76deabf6b/               0
# uncompressed_size
# 1                 0
```

Usually it doesn't make sense to compress an empty folder. However in my usage I'm saving a folder and restoring it later. The folder could be empty or with contents. With the behavior of `utils::zip` I don't need to do anything special. With current `zip::zip` behavior I need to avoid the zip creation when directory is empty, and avoid the restoring when zip is not generated.

If you decided a zip with empty folder should not be generated, at least it should raise some error and do not create the invalid zip file.

**Comment by gaborcsardi** [comment]
Thanks! Will fix!

**Comment by gaborcsardi** [comment]
Was fixed in #5.
